scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1232,16 +1238,47 @@
                     (launch-matched-tasks! matches conn db fenzo mesos-run-as-user pool-name)
                     (update-host-reservations! rebalancer-reservation-atom matched-job-uuids)
                     matched-considerable-jobs-head?))
+                ; Absolute maximum jobs we will consider autoscaling to.
+                {:keys [max-jobs-for-autoscaling autoscaling-scale-factor]} (config/kubernetes)
+                ; The fraction of jobs we tried to match that didn't actually get matched.
+                fraction-unmatched-jobs (if (pos? number-considerable-jobs) (/ (float number-unmatched-jobs) number-considerable-jobs) 0)
+                ; We want to autoscale any unmatched job or
+                ; we want to scale our max-jobs-for-autoscaling by the fraction of the jobs we weren't able to just match.
```

Comment:
The grammar in this sentence is confusing me, how does the part before the "or" relate to the part after the "or"? Maybe this could be re-written to be more clear?